api/notd/collection_processor.py

```diff
@@ -41,7 +42,7 @@ async def retrieve_collection(self, address: str) -> RetrievedCollection:
         retryCount = 0
         while not openseaResponse:
             try:
-                openseaResponse = await self.requester.get(url=f'https://api.opensea.io/api/v1/asset_contract/{address}')
+                openseaResponse = await self.requester.get(url=f'https://api.opensea.io/api/v1/asset_contract/{address}',headers={"X-API-KEY": f"{os.environ['OPENSEA_API_KEY']}"})
```

Comment:
pass the key in the constructor please. see how this is done for the revue key for the manager.py